### PR TITLE
Deprecate buildType in favor of tag

### PR DESCRIPTION
**Pull request by trevor-e**

### CODE REVIEW

File: gradle-plugin/README.md [hunk 1]
```diff
@@ -144,16 +144,16 @@ emerge {
   // ..
 
   size {
-    buildType.set("release") // Build type to use for grouping builds in the Emerge dashboard
+    tag.set("release") // Build type to use for grouping builds in the Emerge dashboard
```
Comment (rbro112) [inline gradle-plugin/README.md:147]: Nit: Update comment to reflect new naming

File: gradle-plugin/README.md [hunk 2]
```diff
@@ -144,16 +144,16 @@ emerge {
...
 ```
 
 ##### Fields
 
-| Field       | Type     | Default   | Description                                                        |
-|-------------|----------|-----------|--------------------------------------------------------------------|
-| `buildType` | `String` | `release` | The build type to use for grouping builds in the Emerge dashboard. |
+| Field | Type     | Default   | Description                                                        |
+|-------|----------|-----------|--------------------------------------------------------------------|
+| `tag` | `String` | `release` | The build type to use for grouping builds in the Emerge dashboard. |
```
Comment (rbro112) [inline gradle-plugin/README.md:156]: Nit: Update description

File: gradle-plugin/README.md [hunk 3]
```diff
@@ -177,7 +177,7 @@ emerge {
   // ..
 
   performance {
-    buildType.set("release") // Build type to use for grouping builds in the Emerge dashboard
+    tag.set("release") // Build type to use for grouping builds in the Emerge dashboard
```
Comment (rbro112) [inline gradle-plugin/README.md:180]: Update comment

File: gradle-plugin/README.md [hunk 4]
```diff
@@ -190,7 +190,7 @@ emerge {
 | Field         | Type     | Default   | Description                                                          |
 |---------------|----------|-----------|----------------------------------------------------------------------|
 | `projectPath` | `String` |           | The relative gradle path from root to the Emerge performance module. |
-| `buildType`   | `String` | `release` | The build type to use for grouping builds in the Emerge dashboard.   |
+| `tag`         | `String` | `release` | The build type to use for grouping builds in the Emerge dashboard.   |
```
Comment (rbro112) [inline gradle-plugin/README.md:193]: Update description

File: gradle-plugin/README.md [hunk 5]
```diff
@@ -210,7 +210,7 @@ emerge {
   // ..
 
   snapshots {
-    buildType.set("snapshots") // Build type to use for grouping builds in the Emerge dashboard
+    tag.set("snapshots") // Build type to use for grouping builds in the Emerge dashboard
```
Comment (rbro112) [inline gradle-plugin/README.md:213]: Same

File: gradle-plugin/README.md [hunk 6]
```diff
@@ -231,7 +231,7 @@ versions to releases & names, see [Android API levels](https://apilevels.com/).
 
 | Field                                  | Type      | Default                           | Description                                                                                       |
 |----------------------------------------|-----------|-----------------------------------|---------------------------------------------------------------------------------------------------|
-| `buildType`                            | `String`  | `release`                         | The build type to use for grouping builds in the Emerge dashboard.                                |
+| `tag`                                  | `String`  | `release`                         | The build tag to use for grouping builds in the Emerge dashboard.                                 |
```
Comment (rbro112) [inline gradle-plugin/README.md:234]: Same

File: gradle-plugin/plugin/src/main/kotlin/com/emergetools/android/gradle/util/network/EmergeUploadRequest.kt [hunk 7]
```diff
@@ -53,6 +54,7 @@ fun fetchSignedUrl(
   val request: Request = Request.Builder().apply {
     url(url)
     post(body)
+    addHeader("User-Agent", "${SOURCE_GRADLE_PLUGIN}_${BuildConfig.VERSION}")
```
Comment (trevor-e) [inline gradle-plugin/plugin/src/main/kotlin/com/emergetools/android/gradle/util/network/EmergeUploadRequest.kt:57]: @rbro112 snuck this in too

Comment (rbro112) [inline gradle-plugin/plugin/src/main/kotlin/com/emergetools/android/gradle/util/network/EmergeUploadRequest.kt:57]: Nice, figured you copied it to user-agent here, but we currently just pass this as source to our API: https://github.com/EmergeTools/emerge-android/blob/6fe18887bc31d35373da411656883e843a622090/gradle-plugin/plugin/src/main/kotlin/com/emergetools/android/gradle/tasks/upload/BaseUploadTask.kt#L153

Comment (rbro112) [reply gradle-plugin/plugin/src/main/kotlin/com/emergetools/android/gradle/util/network/EmergeUploadRequest.kt:57]: Lol you beat my comment https://github.com/EmergeTools/emerge-android/pull/118#discussion_r1499925687

**Review by rbro112 — APPROVED** [review]

**Pull request merged**